server/notifications/user_notifier.go

```diff
@@ -9,33 +9,42 @@ import (
 )
 
 type EmailClient interface {
-	SendEmail(ctx context.Context, email, message string) error
+	SendEmail(ctx context.Context, email, subject, message string) error
 }
 
 type Repository interface {
 	FindAll() ([]models.User, error)
 }
 
+type RateMessageFormatter interface {
+	SetRate(rate *models.Rate)
+	Subject() string
+	String() string
+}
+
 type UsersNotifier struct {
-	mailClient     EmailClient
-	rateFetcher    server.RateFetcher
-	userRepository Repository
+	mailClient       EmailClient
+	rateService      server.RateService
+	userRepository   Repository
+	messageFormatter RateMessageFormatter
 }
 
 func NewUsersNotifier(
 	mailClient EmailClient,
-	rateFetcher server.RateFetcher,
+	rateService server.RateService,
 	userRepository Repository,
+	msgFormatter RateMessageFormatter,
 ) *UsersNotifier {
 	return &UsersNotifier{
-		mailClient:     mailClient,
-		rateFetcher:    rateFetcher,
-		userRepository: userRepository,
+		mailClient:       mailClient,
+		rateService:      rateService,
+		userRepository:   userRepository,
+		messageFormatter: msgFormatter,
 	}
 }
 
 func (n *UsersNotifier) Notify(ctx context.Context) {
-	rate, err := n.rateFetcher.FetchRate("USD", "UAH")
+	rate, err := n.rateService.FetchRate(ctx, "USD", "UAH")
 	if err != nil {
 		slog.Warn("failed to fetch rate", slog.Any("error", err))
```

Comment:
please, be consistent with logging. [Here](https://github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pull/5/files#diff-0b43a75732c50c5a45014e8723fc978abbc2dff69f427d046c8b2a20a9ff9849R81) your log start with the upper-case, but here you do it in a lower-case.

Google style guide suggests making errors not capitalized, but logs capitalized
https://google.github.io/styleguide/go/decisions.html#error-strings

> On the other hand, the style for the full displayed message (logging, test failure, API response, or other UI) depends, but should typically be capitalized.

Reply:
I'll unify to using lowercase logs